### DISCUSSION TITLE
The bundle size of formily is huge - Can we tree-shaking that?

### DISCUSSION THREAD

**Discussion by nhattran998** [alibaba/formily#3211] [Q&A 2.x] [answered]
Thanks for the great library, I'm so excited when using this. But when I check the bundle size, it's equal to react + react-dom sizes. So this is the trade-off when using this vs react-hook-form? Or might be I do not know the way to optimize that, so hopefully, someone can point to me the problem? 🙏 Thank you.
![image](https://user-images.githubusercontent.com/24296018/174461716-4ee24452-b046-489a-bba8-ba2a50cacffb.png)

**Comment by janryWang** [comment, marked as answer]
If you attach great importance to the bundle size, I would recommend you to use react-hook-form. On the contrary, formily pays more attention to the problem of the form itself. To solve so many problems must require more abstraction, but the more abstraction, the larger the bundle size will be. Of course, maybe in the future I will launch formily-lite, However, as far as the current energy is concerned, there is no time to invest

**Reply by nhattran998** [reply]
Thanks for your quick reply. I'm just curious about this 😆.
